R/utils.R

```diff
@@ -98,3 +98,12 @@ is_quantity <- function(x) {
   }
   x
 }
+
+# When we want to use vctrs, but treat lists like purrr does
+vctrs_list_compat <- function(x) {
+  if (is.list(x) && !vec_is_list(x)) {
+    unclass(x)
```

Comment:
I don't think this is quite the right heuristic. It means that rcrd vectors will also be unclassed, and we don't want that.

I guess it can't be just this

```r
if (is.data.frame(x)) {
  unclass(x)
} else {
  x
}
```

because we want scalar lists like `lm` objects to be recyclable and work with `map2()` and friends for backwards compat.

So maybe it is like this?

```r
if (is.data.frame(x) || (is.list(x) && !vec_is(x))) {
  # Treat data frames and S3 scalar lists like bare lists.
  # Ensures rcrd vctrs retain their class.
  unclass(x)
} else {
  x
}
```

Reply:
After we fix #963, we should definitely add some tests for rcrd vectors with `map()`, `map2()`, and `pmap()` to make sure the 3 important cases work as expected